fhir/com.b2international.snowowl.fhir.core/src/com/b2international/snowowl/fhir/core/request/FhirResourceSearchRequest.java

```diff
@@ -229,11 +232,28 @@ protected T toFhirResource(RepositoryContext context, ResourceFragment resource)
 		includeIfFieldSelected(CodeSystem.Fields.TEXT, () -> Narrative.builder().div("<div></div>").status(NarrativeStatus.EMPTY).build(), entry::text);
 		includeIfFieldSelected(CodeSystem.Fields.VERSION, resource::getVersion, entry::version);
 		includeIfFieldSelected(CodeSystem.Fields.PUBLISHER, resource::getOwner, entry::publisher);
-		includeIfFieldSelected(CodeSystem.Fields.COPYRIGHT, resource::getCopyright, entry::copyright);
 		includeIfFieldSelected(CodeSystem.Fields.LANGUAGE, resource::getLanguage, entry::language);
 		includeIfFieldSelected(CodeSystem.Fields.DESCRIPTION, resource::getDescription, entry::description);
 		includeIfFieldSelected(CodeSystem.Fields.PURPOSE, resource::getPurpose, entry::purpose);
 		
+		//includeIfFieldSelected(CodeSystem.Fields.COPYRIGHT, resource::getCopyright, entry::copyright);
+		
+		if (CompareUtils.isEmpty(CodeSystem.Fields.COPYRIGHT) || fields().contains(CodeSystem.Fields.COPYRIGHT)) {
```

Comment:
This if block does not look right. Could you please move the copyright builder method calls to the corresponding model-specific SearchRequest implementations (`FhirCodeSystemSearchRequest`). 🙇